docs/book/src/reference/compiler_intrinsics.md

```diff
@@ -25,20 +25,30 @@ __size_of<T>() -> u64
 ___
 
 ```sway
-__size_of_str<T>() -> u64
+__size_of_str_array<T>() -> u64
```

Comment:
I think this PR needs to be marked `*Breaking*` because of this renaming and because of `_check_str_type` ->
`__assert_is_str_array`.

Reply:
Agree. Done.